openupgradelib/openupgrade.py

```diff
@@ -1056,12 +1060,16 @@ def wrapped_function(cr, version):
                 logger.info(
                     "%s: %s-migration script called with version %s" %
                     (module, stage, version))
+                # Set up useful default context values
+                context = context or {}
+                context['migrate_version'] = version and version or None
+                context['module_name'] = module and module or None
```

Comment:
- "useful" as in "serve a current use case better"? -- As to my knowledge, it isn't.

or

- "useful" as in "potentially enable future use cases"? -- This was the design goal.


Class of future use cases, which this facilitates, are characterized by the need to determine and persist the exact migration context, whereby "exact migration context" is constituted by a hash of `migrate_version` and `module_name`. Only if those are not `None`, this promise can be kept, but I see no other way to strengthen the promise further, so user has to take care of this.

Basically, this enables tagging records or values or columns/fields with the hash and refer to them precisely afterwards.

Some illustrative examples:
- External ids created by migration scripts
- Reliable Field/column prefixing